### PR TITLE
Implemented controller, Mapper, DTOs

### PR DESCRIPTION
**Pull request by kacper-kukula**

### CODE REVIEW

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
+                .toList();
+    }
+
+    @Override
+    public BookDto getByBookId(Long id) {
+        Book book = bookRepository.getByBookId(id);
+
+        if (book == null) {
+            throw new EntityNotFoundException("Book doesn't exist. ID: " + id);
+        }
```
Comment (okuzan) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: make repo return optional to handle this gracefully

File: src/main/java/com/bookstore/controller/impl/BookControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/api/books")
```
Comment (okuzan) [inline src/main/java/com/bookstore/controller/impl/BookControllerImpl.java]: consider extracting this common URL prefix to properties: `server.servlet.context-path=/api`

File: src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -2,16 +2,52 @@
...
+            return entityManager
+                    .createQuery("SELECT b FROM Book b", Book.class)
+                    .getResultList();
+        }
+    }
+
+    @Override
+    public Book getByBookId(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.find(Book.class, id);
```
Comment (okuzan) [inline src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            return Optional.ofNullable(entityManager.find(Book.class, id));
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstore/controller/BookController.java
Comment (nowickimj) [inline src/main/java/com/bookstore/controller/BookController.java:1]: You may remove this interface.

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
     }
 
     @Override
-    public List<Book> findAll() {
-        return null;
+    public BookDto getByBookId(Long id) {
+        Book book = bookRepository.getByBookId(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book doesn't exist. ID: " + id));
+
+        return bookMapper.toDto(book);
```
Comment (nowickimj) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: Please refactor this method to not store the `book` variable and return results immediately - use `Optional.map()` method.

File: src/main/java/com/bookstore/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.bookstore.controller;
+
+import com.bookstore.dto.BookDto;
+import com.bookstore.dto.CreateBookRequestDto;
+import java.util.List;
+
+public interface BookController {
+
```
Comment (nowickimj) [inline src/main/java/com/bookstore/controller/BookController.java:19]: I see that sometimes you put a new line here and sometimes you don't - please make sure that you keep your code consistent.

**Review by nowickimj — CHANGES_REQUESTED** [review]

**Review by nowickimj — APPROVED** [review]
As of my experience, I see new lines after the class declaration very rarely, but you can have it your way as long as you keep it consistent :)

**Pull request merged**